urlparse4/cgurl.pyx

```diff
@@ -145,20 +149,34 @@ class SplitResultNamedTuple(tuple):
 
         cls.__getattr__ = _get_attr
 
-        return tuple.__new__(cls, (
-            slice_component(url, parsed.scheme).lower(),
-            build_netloc(url, parsed),
-            slice_component(url, parsed.path),
-            slice_component(url, parsed.query),
-            slice_component(url, parsed.ref)
-        ))
+        if six.PY2:
```

Comment:
Two points about this bit of code:
- it would be nice to avoid duplication of code between python 2 and 3
- I'm not sure ``unicode`` function is right here, from the docs:  "Create a new Unicode object from the given encoded string. Encoding defaults to the current default string encoding." - I don't think we should rely on default encoding. Also, if we need to do encoding/decoding, it's more efficient to do this in Cython

Reply:
Regarding returning unicode or not, stdlib urlparse functions can return unicode, so to me it looks more sane to always return unicode even on python 2, maybe it will also be easier to support. Although this might require some extra effort when doing scrapy integration, see e.g. https://github.com/scrapy/scrapy/pull/1949#discussion_r61271457